brevets/flask_brevets.py

```diff
@@ -0,0 +1,133 @@
+"""
+Replacement for RUSA ACP brevet time calculator
+(see https://rusa.org/octime_acp.html)
+
+"""
+
+import flask
+from flask import request
+import arrow  # Replacement for datetime, based on moment.js
+import acp_times  # Brevet time calculations
+from brevetsmongo import insert_brevet, retrieve_brevet
+import config
+
+import logging
+
+###
+# Globals
+###
+app = flask.Flask(__name__)
+CONFIG = config.configuration()
+
+###
+# Pages
+###
+
+
+@app.route("/")
+@app.route("/index")
+def index():
+    app.logger.debug("Main page entry")
+    return flask.render_template('calc.html')
+
+
+@app.errorhandler(404)
+def page_not_found(error):
+    app.logger.debug("Page not found")
+    return flask.render_template('404.html'), 404
+
+@app.route("/_insert_b", methods=["POST"])
+def _insert_b():
+    #Work Here
+    app.logger.debug("in insert function")
+    try:
+        # Read the entire request body as a JSON
+        # This will fail if the request body is NOT a JSON.
+        input_json = request.json
+        # if successful, input_json is automatically parsed into a python dictionary!
+
+        # Because input_json is a dictionary, we can do this:
+        brevet_dist = input_json["brevet_dist"]
+        start_time = input_json["start_time"]
+        control_list = input_json["control_list"]
```

Comment:
Code here looks good! One thing worth adding though is a check to make sure the data that's about to be inserted is valid data. One thing I noticed when testing your code is that you can submit as many blank forms as you want, as a result the database gets full of empty documents (apart from the distance and start time). I'll make a suggestion, since I believe that the project calls for this check.

```suggestion
        brevet_dist = input_json["brevet_dist"]
        start_time = input_json["start_time"]
        control_list = input_json["control_list"]
        
        if len(control_list) == 0:
            return flask.jsonify(result={}, message="Nothing to insert!", status=0, mongo_id=None) 
```

This only handles one specific case. It's important to make sure that the data that's being inserted is valid data because we don't want to put invalid data in the database.